conans/test/functional/generators/premake_test.py

```diff
@@ -0,0 +1,77 @@
+import os
+import textwrap
+import unittest
+
+from nose.plugins.attrib import attr
+
+from conans import load
+from conans.client.tools import chdir, which
+from conans.test.utils.tools import TestClient
+
+
+@attr("slow")
+@attr("premake")
+@unittest.skipIf(which("premake5") is None, "Needs premake5")
+class PremakeGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.client = TestClient()
+        self.client.run("new test/1.0")
+        self.client.run("create . danimtb/testing")
+        conanfile = textwrap.dedent("""
+        [requires]
+        test/1.0@danimtb/testing
+        
+        [generators]
+        premake
+        """)
+        premake = textwrap.dedent("""
+        include("conanbuildinfo.lua")
```

Comment:
Why this `include()` is  needed here instead of the `require` indicated in the docs? I tested it with `require "conanbuildinfo.lua"` and it didn't work. Maybe you know it @SSE4

Reply:
[include](https://github.com/premake/premake-4.x/blob/master/src/base/globals.lua#L120): read and execute a lua file (it calls [dofile](https://www.lua.org/manual/5.3/manual.html#pdf-dofile))
[require](https://www.lua.org/manual/5.3/manual.html#pdf-require): load a lua module